### ISSUE TITLE
[HOLD for c+ newdot payment][$1000] Compose box loses focus when reclicking same chat in LHN

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#23676]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Open any chat
2. Notice that the composer is focused
3. Click on the same chat in LHN. Notice the composer loses focus
## Expected Result: 
Compose box don't lose focus when Clicking on the same chat in LHN
## Actual Result: 
Compose box loses focus

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.46-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/1ffdfd08-c54a-4077-bde2-b29a0b1bc807

https://github.com/Expensify/App/assets/43996225/f1fa0c34-d966-4b34-9170-3c4f22d6a536

**Expensify/Expensify Issue URL:**
**Issue reported by:** @daveSeife
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1690317170412039

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01a795f9206baa2f0c</li>
        <li>Upwork Job ID: 1684464091337949184
<li>Last Price Increase: 2023-08-07</li></li>
        2023-07-27<li>Automatic offers: </li>
<ul>
<li>daveSeife | Reporter | 25800239</li>
<li>tienifr | Contributor | 26045509</li>
</ul></li>
    </ul>
</details>

**Comment by kadiealexander** [comment]
Repro:

https://github.com/Expensify/App/assets/59587260/36fc2efe-ff05-4931-8942-71b26726eb39

**Comment by Santhosh-Sellavel** [comment]
@mountiny @hannojg @0xmiroslav This is a regression from #21406, please take care.

**Comment by kadiealexander** [comment]
@0xmiroslav @Santhosh-Sellavel any updates?

**Comment by hannojg** [comment]
Checking …

**Comment by hannojg** [comment]
Every time the current report page would change, the callback would change, and thus the `SidebarLinks` would re-render as well. The effect would be the same as just passing `currentReportID`

Here is what we could do instead:

**Comment by hannojg** [comment]
We can save the `currentReportID` in a ref (and update the ref from a `useEffect`).
Then we create a memoized callback as @tienifr pointed out (minor note: I'd only pass a string as parameter).

For this part 

> We can save the `currentReportID` in a ref (and update the ref from a `useEffect`).

we already have a hook called `usePrevious`. The full solution would roughly look like this:

```jsx
function SidebarLinksData({currentReportID}) {
  const currentReportIDRef = usePrevious(currentReportID);
  const isReportActive = useCallback((reportID) => reportID === currentReportIDRef.current, []);

  return (
    <SidebarLinks {...} isReportActive={isReportActive} />
   );
}
```

@tienifr Feel free to use that to update your proposal 😊

**Comment by Santhosh-Sellavel** [comment]
This should be handled as the Regression, if not I would have reviewed it myself!

cc: @0xmiroslav @kadiealexander

**Comment by kadiealexander** [comment]
@0xmiroslav I thought you were taking care of this issue as per this [comment](https://github.com/Expensify/App/issues/23676#issuecomment-1654665134). Could you please clarify if you're going to be submitting a PR, or just doing C+ duties?

**Comment by kadiealexander** [comment]
@Santhosh-Sellavel @0xmiroslav This hasn't followed proper regression procedures so I'm a little lost. @Santhosh-Sellavel could you please take back the C+ duties to keep this moving forward? We'll just treat it as a new bug.

**Comment by Santhosh-Sellavel** [comment]
@tienifr's [proposal](https://github.com/Expensify/App/issues/23676#issuecomment-1652400735) looks good to me

C+ Reviewed
🎀 👀 🎀 

Please unassign @0xmiroslav & assign me back  @kadiealexander, thanks!

**Comment by kadiealexander** [comment]
@MonilBhavsar please take a look!

**Comment by MonilBhavsar** [comment]
Looks good! 
Good point for not using currentReportID as a dependency. Thanks for the discussion 👍 
Are we treating this report as regression?

**Comment by Santhosh-Sellavel** [comment]
@kadiealexander This issue is on hold for https://github.com/Expensify/App/pull/24619 refer this [comment](https://github.com/Expensify/App/pull/24360#issuecomment-1680448891). Please add Hold to the title thanks!

@tienifr Let us know when the other PR is merged thanks!

**Comment by kadiealexander** [comment]
Thanks @tienifr!!

**Comment by MonilBhavsar** [comment]
Bumping to weekly

**Comment by Santhosh-Sellavel** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

* [x]  [@Santhosh-Sellavel] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/21406
* [x]  [@Santhosh-Sellavel] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/21406#issuecomment-1723931284
* [x]  [@Santhosh-Sellavel] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
* [x]  [@Santhosh-Sellavel] Determine if we should create a regression test for this bug. https://expensify.slack.com/archives/C049HHMV9SM/p1695055888894839
* [x]  [@Santhosh-Sellavel] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again. We should already have this on other issue
* [x]  [@kadiealexander] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by Santhosh-Sellavel** [comment]
@kadiealexander Please leave a payment summary here.

**Comment by kadiealexander** [comment]
Payouts due:

- [ ] Issue Reporter: $250 @daveSeife
- [ ] Contributor: $1000 @tienifr
- [ ] Contributor+: $1000 @Santhosh-Sellavel

Eligible for 50% #urgency bonus? No

Upwork job is [here]().

**Comment by Santhosh-Sellavel** [comment]
Requested on ND

**Comment by kadiealexander** [comment]
Not overdue.

**Comment by Santhosh-Sellavel** [comment]
I got paid lets close this one, thanks!
